src/modules/dashboards/saga.ts

```diff
@@ -698,11 +699,31 @@ export function* cloneDashboard({
     const blobApi = yield getContext(BLOB_API);
 
     const model: DashboardModel = yield blobApi.getDashboardById(dashboardId);
+    const uniqueIdMap = model.widgets.reduce((acc, widget) => {
+      return {
+        ...acc,
+        [widget.id]: createWidgetId(),
+      };
+    }, {});
 
     const uniqueIdWidgets = model.widgets.map((widget) => ({
       ...widget,
-      id: createWidgetId(),
-    }));
+      ...('datePickerId' in widget && {
```

Comment:
Please, add additional test cases to check **unique ids** for filter and date picker widgets/